hibernate-envers/src/main/java/org/hibernate/envers/internal/entities/mapper/relation/ToOneIdMapper.java

```diff
@@ -58,8 +60,16 @@ public boolean mapToMapFromEntity(
 		// bi-directional relation, we always store the "old", unchanged data, to prevent storing changes made
 		// to this field. It is the responsibility of the collection to properly update it if it really changed.
 		Object entity = nonInsertableFake ? oldObj : newObj;
+
+		// fix HHH-13760 - try to aggressively un-proxy this entity to help get the correct type of data later
+		// in mapToMapFromEntity. But it might fail while getImplementation() if object is deleted or other reasons.
+		// We catch the exception and fallback to call mapToMapFromEntity directly with the HibernateProxy entity
 		if ( lazyMapping && entity instanceof HibernateProxy ) {
-			entity = ( (HibernateProxy) entity ).getHibernateLazyInitializer().getImplementation();
+			try {
+				entity = ((HibernateProxy) entity).getHibernateLazyInitializer().getImplementation();
+			} catch ( PersistenceException e ) {
+				// Ignore the exception and fallback to call mapToMapFromEntity directly
```

Comment:
IMO the exception should at least be logged with a DEBUG level. Since this throws a `PersistenceException` I assume that this will cause a transaction rollback due to the semantics of JPA, so this might not be the best way to fix this. Can you verify if the TX is marked as rollback only here?

Reply:
Ok, I just checked the code and the TX is not marked as rollback-only in this case, so this is fine.

If you introduce the logging, I will merge this.